nemo/collections/common/tokenizers/sentencepiece_tokenizer.py

```diff
@@ -105,17 +106,11 @@ def tokens_to_text(self, tokens):
         return self.tokenizer.decode_pieces(tokens)
 
     def ids_to_text(self, ids):
-        text = ""
-        last_i = 0
+        if isinstance(ids, np.ndarray):
+            ids = ids.tolist()
 
-        for i, id in enumerate(ids):
-            if id in self.id_to_special_token:
-                text += self.tokenizer.decode_ids(ids[last_i:i]) + " "
-                text += self.id_to_special_token[id] + " "
-                last_i = i + 1
-
-        text += self.tokenizer.decode_ids(ids[last_i:])
-        return text.strip()
+        ids_ = [id_ for id_ in ids if id_ not in self.id_to_special_token]
+        return self.tokenizer.decode_ids(ids_)
```

Comment:
Does this return the same thing?

Reply:
The update still returns decoded text, just with special tokens removed. This makes it consistent with the YTTM and word tokenizers. In general, the way we're handling special tokens in sentencepiece wrapper is kind of dicey (as it is dependent on the order of the passed tokens), and I believe we might want to take some other route for it.

Reply:
is this also consistent with autotokenizer? does that also remove special tokens?

Reply:
Yes, autotokenizer removes the special tokens too.